WordPress/src/main/java/org/wordpress/android/ui/mysite/ImprovedMySiteFragment.kt

```diff
@@ -153,190 +86,32 @@ class ImprovedMySiteFragment : Fragment(),
 
         actionable_empty_view.button.setOnClickListener { viewModel.onAddSitePressed() }
 
-        val layoutManager = LinearLayoutManager(activity)
-
-        savedInstanceState?.getParcelable<Parcelable>(KEY_LIST_STATE)?.let {
-            layoutManager.onRestoreInstanceState(it)
-        }
-
-        recycler_view.layoutManager = layoutManager
-
-        val adapter = MySiteAdapter(imageManager, uiHelpers)
-
-        savedInstanceState?.getBundle(KEY_NESTED_LISTS_STATES)?.let {
-            adapter.onRestoreInstanceState(it)
-        }
-
-        recycler_view.adapter = adapter
-
         viewModel.uiModel.observe(viewLifecycleOwner, {
             it?.let { uiModel ->
                 loadGravatar(uiModel.accountAvatarUrl)
                 when (val state = uiModel.state) {
-                    is State.SiteSelected -> loadData(state.items)
+                    is State.SiteSelected -> loadData()
                     is State.NoSites -> loadEmptyView(state.shouldShowImage)
                 }
             }
         })
-        viewModel.onBasicDialogShown.observe(viewLifecycleOwner, {
-            it?.getContentIfNotHandled()?.let { model ->
-                dialogViewModel.showDialog(requireActivity().supportFragmentManager,
-                        BasicDialogModel(
-                                model.tag,
-                                getString(model.title),
-                                getString(model.message),
-                                getString(model.positiveButtonLabel),
-                                model.negativeButtonLabel?.let { label -> getString(label) },
-                                model.cancelButtonLabel?.let { label -> getString(label) }
-                        ))
-            }
-        })
-        viewModel.onTextInputDialogShown.observe(viewLifecycleOwner, {
-            it?.getContentIfNotHandled()?.let { model ->
-                val inputDialog = TextInputDialogFragment.newInstance(
-                        getString(model.title),
-                        model.initialText,
-                        getString(model.hint),
-                        model.isMultiline,
-                        model.isInputEnabled,
-                        model.callbackId
-                )
-                inputDialog.setTargetFragment(this, 0)
-                inputDialog.show(parentFragmentManager, TextInputDialogFragment.TAG)
-            }
-        })
-        viewModel.onQuickStartMenuShown.observe(viewLifecycleOwner, {
-            it?.getContentIfNotHandled()?.let { id ->
-                ((parentFragmentManager.findFragmentByTag(id) as? QuickStartMenuFragment)
-                        ?: QuickStartMenuFragment.newInstance(id))
-                        .show(parentFragmentManager, id)
+        viewModel.onSiteChange.observe(viewLifecycleOwner, {
```

Comment:
⚠️ Tbh I haven't noticed any slowness when I tested the changes on my device + emulator. However, I'm wondering if they might be caused by the way LiveData behave -> we start re-creating the content fragment after onStart, because the LiveData won't receive any updates until then -> perhaps using a different approach which would start the process sooner would speed things up. Wdyt?